### PR TITLE
fix connection reset by peer case

### PR DESCRIPTION
**Pull request by d-a-v**
... where pcb is set to null in ClientContext::_error but not reported to WiFiClient
also rename functions *_sent to *_acked (:sent to :ack in debug)

### CODE REVIEW

File: libraries/ESP8266WiFi/src/include/ClientContext.h [hunk 1]
```diff
@@ -536,7 +536,7 @@ class ClientContext
         tcp_sent(_pcb, NULL);
         tcp_recv(_pcb, NULL);
         tcp_err(_pcb, NULL);
-        _pcb = NULL;
+        _pcb = 0;
```
Comment (devyte) [inline libraries/ESP8266WiFi/src/include/ClientContext.h]: _pcb is a pointer, so assignment was current with NULL.

Comment (d-a-v) [reply libraries/ESP8266WiFi/src/include/ClientContext.h]:
```
$ grep '_pcb =' ClientContext.h
            _pcb = 0;
            _pcb = 0;
        _pcb = NULL;
```
It is for coherency. 0 is c++ way for NULL (I prefer NULL myself). Should I change them all ?

Comment (devyte) [reply libraries/ESP8266WiFi/src/include/ClientContext.h]: I didn't realize, but yes please.

Comment (devyte) [reply libraries/ESP8266WiFi/src/include/ClientContext.h]: Oh, yes, for some reason I thought of ClientContext as C code. Nullptr please!

**Comment by d-a-v** [comment]
Changes done but PR garbled, I will fix it clean.

**Pull request merged**